app/hooks/useFirebase.js

```diff
@@ -0,0 +1,54 @@
+import { useState, useRef } from "react";
+import { PhoneAuthProvider, signInWithCredential } from "firebase/auth";
+import { auth } from "../../firebase/firebase";
+
+const useFirebase = () => {
+  const [phoneNumber, setPhoneNumber] = useState();
+  const [verificationCode, setVerificationCode] = useState();
+  const [verificationId, setVerificationId] = useState();
+  const [message, showMessage] = useState();
```

Comment:
[nit pick] I personally feel the `callback` in `useState` should start with `set...` so this line would be (if you agree with this suggestion, you'll need to update lines 19, 21, 33, & 37): 
```suggestion
  const [message, setShowMessage] = useState();
```